code/game/machinery/suit_dispenser.dm

```diff
@@ -2,120 +2,179 @@
+	dispenser_flags &= ~SD_BUSY
 
+	var/turf/T = get_turf(src)
+	if(!(S && T)) // in case we got destroyed while we slept
+		return 1
+	for(var/O in S.to_spawn)
+		new O(T)
+	if(emagged)
+		emagged = FALSE
+	if(greet && user && !user.stat) // in case we got destroyed while we slept
+		to_chat(user,"<span class='notice'>[S.name] specialisation processed. Have a good day.</span>")
 
 
-/obj/machinery/suit_dispenser/ert
-	desc = "An industrial U-Tak-It Dispenser unit designed to fetch all kinds of space suits. This one distribributes Emergency Responder space suits."
-
-/obj/machinery/suit_dispenser/ert/attack_hand(var/mob/user)
-	if(sentStrikeTeams(TEAM_ERT))
-		var/datum/striketeam/ert/team = sent_strike_teams[TEAM_ERT]
-		if (user in response_team_members)
-			if (user in distributed_ert_suits)
-				to_chat(user,"<span class='warning'>You've already picked up your suit.</span>")
-			else
-				if (user.key == team.leader_key)
-					to_chat(user,"<span class='notice'>Identified as [user.real_name]. Here is your suit commander. Have a good day.</span>")
-
-					flick("suitdispenser-flick",src)
-
-					sleep(17)
-
-					if (user in distributed_ert_suits)
-						return
-
-					distributed_ert_suits |= user
-
-					var/turf/T = get_turf(src)
-
-					new /obj/item/clothing/suit/space/ert/commander(T)
-					new /obj/item/clothing/head/helmet/space/ert/commander(T)
-
-				else
-					to_chat(user,"<span class='notice'>Identified as [user.real_name]. Please choose your suit specialization.")
-
-					var/list/suit_list = list(
-						"Security",
-						"Medical",
-						"Engineer",
-						"CANCEL"
-						)
+/obj/machinery/suit_dispenser/emag(var/mob/user)
+	if(!emagged)
+		emagged = TRUE
+		if(user)
+			user.visible_message("<span class='warning'>\The [name] slides a weird looking ID into \the [src], and sparks come flying out!</span>","<span class='warning'>You temporarily short the safety mechanisms.</span>")
+		playsound(get_turf(src), pick(spark_sound), 75, 1)
+		spark(src,4)
 
-					var/choice = input("Choose your suit specialization.", "Suit Dispenser") in suit_list
+/obj/machinery/suit_dispenser/proc/promptfornum(var/mob/user)
+	var/num = input(user,"How many supplies would you like to add? (Set to -1 for infinite)","Suit Dispenser", null) as num|null
+	if(num == -1)
+		dispenser_flags |= SD_UNLIMITED
+	else return num
 
-					if(choice == "CANCEL")
-						return
+////////////////////////////// ERT SUIT DISPENSERS ///////////////////////////
 
-					flick("suitdispenser-flick",src)
 
-					sleep(17)
+/datum/suit/ert/security
+	name = "Security"
+	to_spawn = list(/obj/item/clothing/suit/space/ert/security,/obj/item/clothing/head/helmet/space/ert/security)
 
-					if (user in distributed_ert_suits)
-						return
+/datum/suit/ert/medical
+	name = "Medical"
+	to_spawn = list(/obj/item/clothing/suit/space/ert/medical,/obj/item/clothing/head/helmet/space/ert/medical)
 
-					distributed_ert_suits |= user
+/datum/suit/ert/engineer
+	name = "Engineer"
```

Comment:
would make more sense as "Engineering" since that way it will parse as "there are no more engineering suits left" instead of "there are no more engineer suits left"

for the same reason you might want to make all of these names improper instead of randomly capitalizing the department name